### ISSUE TITLE
[REGRESSION][PC][bazzite-deck] Graphics corruption under gaming mode since 20240224

### ISSUE THREAD

**Issue by d3-X-t3r** [ublue-os/bazzite#824]
### Describe the bug

Graphics under gaming mode is corrupted on bazzite-deck since build 20240224, and all subsequent builds.

This affects both Steam itself, and all games subsequently launched (screenshots: [[1]](https://cdn.discordapp.com/attachments/1087140957096517672/1212627415830888518/20240229_180356.jpg?ex=65f28649&is=65e01149&hm=61ad3f44520359ba9543c91888ec79ebe294011d8d26b29bb9ff2688020aaa51&) and [[2]](https://cdn.discordapp.com/attachments/1087140957096517672/1212634291993837628/20240229_180910.jpg?ex=65f28cb0&is=65e017b0&hm=f6f58173e4604b17aca5f189cdd5872c2b28cdb8c052df7aa0da7c5e75b9e7fc&) ). 

However, games launched from Steam via the desktop (KDE) work fine, even when launching them using the `gamescope` command option in steam - so it appears that only gaming mode is affected.

Builds 20240223 and earlier (20240220 etc) are all fine. 
These were the changes when upgraded to 20240224:

```
  discover-overlay 0.6.9-1.fc39 -> 0.7.0-1.fc39
  tuned 2.22.0.ublue.0.0.git.1331.8483cdb3-1.fc39 -> 2.22.1.ublue.0.0.git.1335.8ff17bce-1.fc39
  tuned-gtk 2.22.0.ublue.0.0.git.1331.8483cdb3-1.fc39 -> 2.22.1.ublue.0.0.git.1335.8ff17bce-1.fc39
  tuned-ppd 2.22.0.ublue.0.0.git.1331.8483cdb3-1.fc39 -> 2.22.1.ublue.0.0.git.1335.8ff17bce-1.fc39
  tuned-profiles-atomic 2.22.0.ublue.0.0.git.1331.8483cdb3-1.fc39 -> 2.22.1.ublue.0.0.git.1335.8ff17bce-1.fc39
  tuned-profiles-cpu-partitioning 2.22.0.ublue.0.0.git.1331.8483cdb3-1.fc39 -> 2.22.1.ublue.0.0.git.1335.8ff17bce-1.fc39
  tuned-utils 2.22.0.ublue.0.0.git.1331.8483cdb3-1.fc39 -> 2.22.1.ublue.0.0.git.1335.8ff17bce-1.fc39
  ublue-update 1.1.3+8a926c5-1.fc39 -> 1.2.3+1e670f0-1.fc39
```

This is on a Zen 3+ / Radeon 680M APU. Full system details in the Hardware section below.

### What did you expect to happen?

Graphics should not be corrupted.

### Output of `rpm-ostree status`

```shell
State: idle
Deployments:
  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck:39-20240223
                   Digest: sha256:07073dc8dfb92a470209d30329c50d50675c27d26359736cc0593c6cd97bc804
                  Version: 39.20240223.0 (2024-02-23T16:58:54Z)
                     Diff: 54 downgraded, 2 removed, 2 added
          LayeredPackages: gnome-boxes libvirt libvirt-client libvirt-daemon-config-network libvirt-nss virt-manager
                Initramfs: '-I /etc/crypttab /etc/modprobe.d/amdgpu.conf /etc/modprobe.d/deck-blacklist.conf' 
                   Pinned: yes

  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck:39-20240224
                   Digest: sha256:3dd38fb5fc1c847a8ac2d3716ca1beb4c7efeb213e3e78b720015ce5fbb86987
                  Version: 39.20240223.0 (2024-02-24T00:48:19Z)
          LayeredPackages: gnome-boxes libvirt libvirt-client libvirt-daemon-config-network libvirt-nss virt-manager
                Initramfs: '-I /etc/crypttab /etc/modprobe.d/amdgpu.conf /etc/modprobe.d/deck-blacklist.conf' 
                   Pinned: yes

● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck:latest
                   Digest: sha256:4151c51fe4e5e394de97987a47600a09b0866a38569baeadccd88b146ea9875c
                  Version: 39.20240224.0 (2024-02-25T07:25:06Z)
          LayeredPackages: gnome-boxes libvirt libvirt-client libvirt-daemon-config-network libvirt-nss virt-manager
                Initramfs: '-I /etc/crypttab /etc/modprobe.d/amdgpu.conf /etc/modprobe.d/deck-blacklist.conf' 
                   Pinned: yes
```

### Hardware

**Machine**:  Lenovo ThinkPad Z13 Gen 1
**CPU**: AMD Ryzen 7 PRO 6850U
**GPU**: Radeon 680M (RDNA 2 / Rembrandt)
**Display**: Integrated (2880x1800~60Hz)

**Comment by jpwchang** [comment]
I just wanted to weigh in to confirm that I've also been seeing this issue since the 20240224 update (bazzite-deck-gnome on a Steam Deck LCD). At first I was terrified that it was a hardware failure, but several observations point to it being a software issue instead:

1. First, as the original post above states, the issue seems to be specific to game mode - I've never seen it happen when launching games from desktop mode.
2. Second, the visual corruption appears to be tied to individual processes. That is, if the visual corruption appears within a game, fully exiting the game and restarting it (hence spawning a new process) usually makes the issue go away.
3. Related to the above, the issue (which, to be clear, is sporadic) seems to either affect a process from the moment of startup, or not at all. That is, I've never had the visual corruption pop up in the middle of gameplay - if a process is affected, the visual corruption will be there from the moment the process starts. As an aside, it appears that while the corruption seems to be triggered once a game starts, it can randomly affect either the game process itself or the entire steam process. In the latter case, the corruption will of course persist after quitting the game (since the steam process is the one affected), but can be cleared by restarting steam (either by rebooting, switching to desktop mode then back, or using the "restart steam" menu option that is made available through developer mode).

I'm hoping that these observations, alongside all the data in the original report, can be of use with debugging. I might anticipate this being a hard one to debug since it seems pretty inconsistent...

**Comment by d3-X-t3r** [comment]
So the issue appears to have been resolved with the latest stable (20240303). Not that I've tested extensively mind you, but I've rebooted a few times, played a couple of games in game mode, and it all seems to work fine. I'll continue my testing for a bit and will close the ticket if it's all good now.

**Comment by KyleGospo** [comment]
> So the issue appears to have been resolved with the latest stable (20240303). Not that I've tested extensively mind you, but I've rebooted a few times, played a couple of games in game mode, and it all seems to work fine. I'll continue my testing for a bit and will close the ticket if it's all good now.

Very odd, no kernel updates or gamescope updates during this time frame, would love to know exactly what broke.
